_projects/2023/sunpy/sunkit_image.md

```diff
@@ -0,0 +1,56 @@
+---
+name: sunkit-image and integration with sunpy
+desc: Replace the sphinx base theme with a newer version.
+requirements:
+ - Experience with Python.
+ - Experience with sunpy Map.
+difficulty: medium to low
+issues:
+ - https://github.com/sunpy/sunpy-sphinx-theme/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc
+mentors:
+ - nabobalis
+ - YOUR NAME HERE
+initiatives:
+ - GSOC
+project_size:
+ - 175 h
+tags:
+ - sunpy
+ - Python
+ - refactoring
+collaborating_projects:
+ - sunpy
+---
+
+#### Description
+
+In a previous GSoC project, [sunkit-image](https://github.com/sunpy/sunkit-image), a image analysis Python library for solar physics was created.
+Functions were created to take arrays and return arrays and this was a big big oversight at the time.
+The goal of the SunPy Project is to ensure interoperability between the packages and this entails being able to pass in a sunpy.map.Map into a function and getting a sunpy.map.Map out.
+
+Therefore the aim of this GSoC project is:
+
+1. Refactor each function to accept and return a sunpy.map.Map instead of a numpy array.
+
```

Comment:
I think we should have some additional aims here.

Reply:
Sure. All ideas welcome.

@wafels had some in the meeting as well.

Reply:
I wonder if all the compensation for solar rotation stuff should be removed from the co-alignment code? Solar rotation in SunPy is now much more mature than it was when the co-alignment code was written. I think if I was doing this now I would perform the solar rotation first and then do the co-alignment. I think a workflow, or a few potential workflows, should be worked out to guide what functionality is needed.

Reply:
Another idea I had was to perform image comparison using different metrics other than cross-correlation - see chapter two of DOI 10.1007/978-1-4471-2458-0.